File: DataStructures/src/main/java/com/zhangdd/queue/ArrayQueue.java
package com.zhangdd.queue;

/**
 * @author zhangdd on 2021/9/23
 */
public class ArrayQueue {

    //表示数组最大容量
    private final int maxSize;

    //队列头
    private int front = -1;

    //队列尾
    private int rear = -1;

    //存放数据的队列
    private final int[] arr;

    public ArrayQueue(int maxSize) {
        this.maxSize = maxSize;
        arr = new int[maxSize];
    }

    private boolean isFull() {
        return rear == maxSize - 1;
    }

    private boolean isEmpty() {
        return front == rear;
    }

    public void addQueue(int n) {
        if (isFull()) {
            System.out.println("当前队列已满，不能添加数据");
            return;
        }
        rear++;
        arr[rear] = n;
    }

    public int getQueue() {
        if (isEmpty()) {
            throw new RuntimeException("当前队列为空，没有数据");
        }
        front++;
        return arr[front];
    }

    public void showQueue() {
        if (isEmpty()) {
            System.out.println("当前队列为空，没有数据");
            return;
        }
        for (int i = 0; i < arr.length; i++) {
            System.out.printf("arr[%d]=%d\n", i, arr[i]);
        }
    }

    public int headQueue() {
        if (isEmpty()) {
            throw new RuntimeException("当前队列为空，没有数据");
        }
        return arr[front + 1];
    }

}

File: python/DataStructures/sparsearry/ArrayQueue.py
#!/usr/bin/env python3
"""
@author: zhangdd on 2022/1/5
数组实现队列
"""


class Queue:
    def __init__(self, max_size=10):
        self.max_size = max_size
        self.rear = -1
        self.front = -1
        self.queue = [None] * max_size

    def push(self, num):
        if self.rear == self.max_size - 1:
            print('当前队列已满')
            return
        else:
            self.rear += 1
            self.queue[self.rear] = num

    def pop(self):
        if self.front < self.rear:
            self.front += 1;
            print(self.queue[self.front])
        else:
            print('当前队列为空')


queue = Queue(10)
command = 0
while True:
    print('1:push data')
    print('2:pop data')
    print('3:exit program')
    command = int(input('please input command'))
    if command == 1:
        num = input('please input num')
        queue.push(num)
    elif command == 2:
        queue.pop()
    elif command == 3:
        break
    else:
        print('input error')
